arangod/Agency/Supervision.cpp

```diff
@@ -1258,6 +1258,26 @@ void Supervision::readyOrphanedIndexCreations() {
 
 void Supervision::enforceReplication() {
   _lock.assertLockedByCurrentThread();
+
+  // First check the number of AddFollower and RemoveFollower jobs in ToDo:
+  auto todos = _snapshot.hasAsChildren(toDoPrefix).first;
+  int nrAddRemoveJobsInTodo = 0;
+  for (auto it = todos.begin(); it != todos.end(); ++it) {
+    auto jobNode = *(it->second);
+    auto t = jobNode.hasAsString("type");
+    if (t.second && (t.first == "addFollower" || t.first == "removeFollower")) {
+      ++nrAddRemoveJobsInTodo;
```

Comment:
Maybe we should define a `maxAddRemoveJobsInTodo` and then early exit here if `nrAddRemoveJobsInTodo` is already bigger.

Reply:
Or return immediately...

Reply:
Done